disdrodb/data_transfer/upload_data.py

```diff
@@ -0,0 +1,261 @@
+    ----------
+    metadata_fpath: str
+        Metadata file path.
+    files_compression: str
+        Compression method for individual files inside station data.
+    """
+
+    data_path = metadata_fpath.replace("metadata", "data")
+    data_path = os.path.splitext(data_path)[0]  # remove trailing ".yml"
+    temp_zip_path = _zip_dir(data_path, files_compression)
+
+    return temp_zip_path
+
+
+def _update_metadata_with_zenodo_url(
+    metadata_fpath: str, deposition_id: int, remote_path: str, sandbox: bool = False
+) -> None:
+    """Update metadata with Zenodo zip file url.
+
+    Parameters
+    ----------
+    metadata_fpath: str
+        Metadata file path.
+    deposition_id: int
+        Zenodo deposition id.
+    remote_path: str
+        Remote path of the zip file.
+    sandbox: bool
+        If True, set reference to Zenodo sandbox for testing purposes.
+    """
+    zenodo_host = "sandbox.zenodo.org" if sandbox else "zenodo.org"
+    metadata_dict = _read_yaml_file(metadata_fpath)
+    metadata_dict["data_url"] = f"https://{zenodo_host}/record/{deposition_id}/files/{remote_path}.zip"
+    _write_yaml_file(metadata_dict, metadata_fpath)
+
+
+def upload_disdrodb_archives(
+    platform: Optional[str] = None,
+    files_compression: Optional[str] = None,
+    force: bool = False,
+    **kwargs,
+) -> None:
+    """Find all stations containing local data and upload them to a remote repository.
+
+    Parameters
+    ----------
+    platform: str, optional
+        Name of the remote platform.
+        If not provided (None), the default platform is Zenodo.
+        The default is platform=None.
+    files_compression: str, optional
+        Compression method for individual files inside station data.
+        The default is files_compression=None.
+    force: bool, optional
+        If True, upload even if data already exists on another remote location.
+        The default is force=False.
+
+    Other Parameters
+    ----------------
+    disdrodb_dir: str, optional
+        DisdroDB data folder path.
+        Must end with DISDRODB.
+    data_sources: str or list of str, optional
+        Data source folder name (eg: EPFL).
+        If not provided (None), all data sources will be uploaded.
+        The default is data_source=None.
+    campaign_names: str or list of str, optional
+        Campaign name (eg:  EPFL_ROOF_2012).
+        If not provided (None), all campaigns will be uploaded.
+        The default is campaign_name=None.
+    station_names: str or list of str, optional
+        Station name.
+        If not provided (None), all stations will be uploaded.
+        The default is station_name=None.
+    """
+
+    metadata_fpaths = get_list_metadata(
+        **kwargs,
+        with_stations_data=True,
+    )
+
+    if not force:
+        metadata_fpaths = _filter_already_uploaded(metadata_fpaths)
+
+    if len(metadata_fpaths) == 0:
+        print("There is no data fulfilling the criteria.")
+        return
```

Comment:
Maybe let's test also this ... Just provide empty directory ...

Reply:
ok